### PR TITLE
[snmp] Fix memory leak in SNMP

### PR DESCRIPTION
**Pull request by fab-kis**
When receiving an async request it is placed in a pending request queue. Not canceling it leads to infinty life cycle for those requests, filling up memory.

### CODE REVIEW

**Comment by fab-kis** [comment]
This PR fixes #8622.

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java [hunk 1]
```diff
@@ -175,6 +176,15 @@ public void onResponse(@Nullable ResponseEvent event) {
         if (event == null) {
             return;
         }
+
+        if (event.getSource() instanceof org.snmp4j.Snmp) {
```
Comment (J-N-K) [inline bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java:180]:
```suggestion
        if (event.getSource() instanceof Snmp) {
```

**Review by J-N-K — CHANGES_REQUESTED** [review]
Otherwise ok

File: bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java (hunk 1 above)
Comment (fab-kis) [reply bundles/org.openhab.binding.snmp/src/main/java/org/openhab/binding/snmp/internal/SnmpTargetHandler.java]: Oh, i miss this, sorry for that.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**